File: Zamba/zambawindows2013/Zamba2010/Zamba.Insert/ZambaInsert/Zamba.RemoteInsert/Form1.vb
Imports Zamba.Servers
Imports System.Text
Imports System.Data.SqlClient
Imports System.IO

''' -----------------------------------------------------------------------------
''' Project	 : Zamba.RemoteInsert
''' Class	 : Form
''' -----------------------------------------------------------------------------
''' <summary>
''' Formulario que sirve para insertar varias filas en las tablas RemoteInsert y DocumentsIndexs
''' </summary>
''' <remarks>
''' </remarks>
''' <history>
''' 	[Gaston]	02/03/2009	Created
''' </history>
''' -----------------------------------------------------------------------------

Public Class Form1

    Dim bnBan As Boolean = True

    Private Sub btnAccept_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAccept.Click

        Dim counter As Integer = 0

        btnCancelProcess.Enabled = True

        Dim documentName As String = "Test"
        Dim docTypeId As String = "7216"
        Dim fileExtension As String = String.Empty
        Dim status As Integer = 3

        Dim docId As Integer = Nothing

        Dim indexId As Integer = 160
        Dim indexValue As String = "prueba"

        Dim fs As New FileStream(Application.StartupPath & "\Prueba.txt", FileMode.OpenOrCreate, FileAccess.Read)
        Dim br As New BinaryReader(fs)
        br.BaseStream.Position = 0
        Dim BinaryImage As Byte() = br.ReadBytes(CInt(br.BaseStream.Length))
        Dim SqlServerParameter As SqlParameter = New SqlParameter("@Blob", BinaryImage)
        SqlServerParameter.DbType = DbType.Binary
        Dim parameters As IDbDataParameter = DirectCast(SqlServerParameter, IDbDataParameter)

        Dim queryRemoteInsert As New StringBuilder
        queryRemoteInsert.Append("INSERT INTO RemoteInsert ")
        queryRemoteInsert.Append("(DocumentName, DocTypeId, SerializedFile, FileExtension, Status) ")
        queryRemoteInsert.Append("VALUES ('")
        queryRemoteInsert.Append(documentName)
        queryRemoteInsert.Append("',")
        queryRemoteInsert.Append(docTypeId)
        queryRemoteInsert.Append(",@Blob,'")
        queryRemoteInsert.Append(fileExtension)
        queryRemoteInsert.Append("',")
        queryRemoteInsert.Append(status)
        queryRemoteInsert.Append(")")

        While (bnBan = True)

            ' Cada cien filas
            If (counter = 100) Then
                ' Se duerme el programa por un segundo
                Threading.Thread.Sleep(1000)
                counter = 0
            End If

            Try
                Server.Con.ExecuteNonQuery(CommandType.Text, queryRemoteInsert.ToString(), parameters)
            Catch ex As Exception
                MessageBox.Show(ex.Message, "Error al insertar en RemoteInsert")
                Exit Sub
            End Try

            docId = getLastInsertedId().ToString()

            If Not (IsNothing(docId)) Then

                Dim queryDocumentsIndexs As New StringBuilder
                queryDocumentsIndexs.Append("INSERT INTO DocumentsIndexs(Id, IndexId, IndexValue) ")
                queryDocumentsIndexs.Append("VALUES (")
                queryDocumentsIndexs.Append(docId)
                queryDocumentsIndexs.Append(",")
                queryDocumentsIndexs.Append(indexId)
                queryDocumentsIndexs.Append(",'")
                queryDocumentsIndexs.Append(indexValue)
                queryDocumentsIndexs.Append("')")

                Try
                    Server.Con.ExecuteNonQuery(CommandType.Text, queryDocumentsIndexs.ToString())
                Catch ex As Exception
                    MessageBox.Show(ex.Message, "Error al insertar en DocumentsIndexs")
                    Exit Sub
                End Try

                queryDocumentsIndexs = Nothing

            Else
                MessageBox.Show("Ha ocurrido un error al generar el temporaryId")
                Exit Sub
            End If

            counter = counter + 1

        End While

    End Sub

    Private Sub btnCancelProcess_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancelProcess.Click
        btnCancelProcess.Text = "Proceso Cancelado"
        bnBan = False
    End Sub

    Private Function getLastInsertedId() As Integer

        Dim value As Integer = 0

        Dim returnId As Object = Server.Con.ExecuteScalar(CommandType.Text, "select max(temporaryId) as 'Id' from remoteinsert")

        If Not (IsNothing(returnId)) Then
            Return (CInt(returnId))
        End If

        Return (Nothing)

    End Function

End Class

File: Zamba/zambawindows2013/Zamba2010/Zamba.WebServices/TestInsertImage/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;
using System.Data.SqlClient;
using System.Data.OracleClient;

namespace TestInsertImage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btInsertIndex_Click(object sender, EventArgs e)
        {
            try
            {
                StringBuilder QueryBuilder = new StringBuilder();

                QueryBuilder.Append("INSERT INTO DocumentsIndexs(Id, IndexId, IndexValue) ");
                QueryBuilder.Append("VALUES (");
                QueryBuilder.Append(txtDocId.Text);
                QueryBuilder.Append(",");
                QueryBuilder.Append(txtIndexId.Text);
                QueryBuilder.Append(",'");
                QueryBuilder.Append(txtIndexValue.Text);
                QueryBuilder.Append("')");

                Zamba.Servers.Server.get_Con(false, false, false).ExecuteNonQuery(CommandType.Text, QueryBuilder.ToString());
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        /// <summary>
        /// Evento que se ejecuta al presionar el botón "Insertar Documento"
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        /// <history>
        ///     [Gaston]    26/05/2009  Modified      Funcionalidad para Oracle
        /// </history>
        private void btInsert_Click(object sender, EventArgs e)
        {
            try
            {
                if (!File.Exists(txtPath.Text))
                {
                    MessageBox.Show("Archivo inexistente");
                    return;
                }

                #region Query

                StringBuilder QueryBuilder = new StringBuilder();
                QueryBuilder.Append("INSERT INTO RemoteInsert ");

                if(Zamba.Servers.Server.isSQLServer)
                {
                    QueryBuilder.Append("(DocumentName, DocTypeId, SerializedFile, FileExtension) ");
                    QueryBuilder.Append("VALUES('");
                }
                else if (Zamba.Servers.Server.isOracle)
                {
                    QueryBuilder.Append("(TemporaryID, DocumentName, DocTypeId, SerializedFile, FileExtension) ");
                    QueryBuilder.Append("VALUES(");
                    QueryBuilder.Append(":TempID,'");
                }
                
                QueryBuilder.Append(txtDocumentName.Text);
                QueryBuilder.Append("',");
                QueryBuilder.Append(txtDocTypeId.Text);

                if(Zamba.Servers.Server.isSQLServer)       
                    QueryBuilder.Append(",@Blob,'");
                else                                       
                    QueryBuilder.Append(",:Blob,'");
                
                FileInfo a = new FileInfo(txtPath.Text);
                QueryBuilder.Append(a.Extension);

                QueryBuilder.Append("')");
                #endregion

                FileStream fs = new FileStream(txtPath.Text, FileMode.Open, FileAccess.Read);
                BinaryReader br = new BinaryReader(fs);

                br.BaseStream.Position = 0;

                byte[] BinaryImage = br.ReadBytes((int)br.BaseStream.Length);

                if (Zamba.Servers.Server.isSQLServer)
                {
                    SqlParameter SqlServerParameter = new SqlParameter("@Blob", BinaryImage);
                    SqlServerParameter.DbType = DbType.Binary;

                    IDbDataParameter[] Parameters = { (IDbDataParameter)SqlServerParameter };
                    Zamba.Servers.Server.get_Con(false, false, false).ExecuteNonQuery(CommandType.Text, QueryBuilder.ToString(), Parameters);
                }
                else if (Zamba.Servers.Server.isOracle)
                {
                    object tempId = (object)Zamba.Servers.Server.get_Con(false, false, false).ExecuteScalar(CommandType.Text, "SELECT MAX(TEMPORARYID) AS LastId FROM REMOTEINSERT");
                    OracleParameter oracleParameter = new OracleParameter();
                    oracleParameter.ParameterName = ":TempID";
                    oracleParameter.DbType = DbType.Int32;

                    if (tempId is DBNull)
                        oracleParameter.Value = 1;
                    else
                        oracleParameter.Value = Convert.ToInt32(tempId.ToString()) + 1;
                    
                    OracleParameter oracleServerParameter = new OracleParameter(":Blob", BinaryImage);
                    oracleServerParameter.DbType = DbType.Binary;

                    IDbDataParameter[] Parameters = { (IDbDataParameter)oracleServerParameter, (IDbDataParameter)oracleParameter };

                    Zamba.Servers.Server.get_Con(false, false, false).ExecuteNonQuery(CommandType.Text, QueryBuilder.ToString(),Parameters);
                }

                txtDocId.Text = GetLastInsertedId().ToString();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        /// <summary>
        /// Método que se utiliza para obtener el máximo temporaryID de la tabla "RemoteInsert"
        /// </summary>
        /// <returns></returns>
        /// <history>
        ///     [Gaston]    26/05/2009  Modified      Funcionalidad para Oracle
        /// </history>
        private static Int32 GetLastInsertedId()
        {
            Int32 Value = 0;
            Object ReturnId = null;

            if(Zamba.Servers.Server.isSQLServer)
                ReturnId = Zamba.Servers.Server.get_Con(false, false, false).ExecuteScalar(CommandType.Text, "select max(temporaryId) as 'Id' from remoteinsert");
            else if (Zamba.Servers.Server.isOracle)
                ReturnId = Zamba.Servers.Server.get_Con(false, false, false).ExecuteScalar(CommandType.Text, "select max(temporaryId) as Id from remoteinsert");

            if (null != ReturnId)
                Int32.TryParse(ReturnId.ToString(), out Value);

            return Value;
        }

        Zamba.Core.RemoteInsert RI = new Zamba.Core.RemoteInsert();
        private void btnUpdateState_Click(object sender, EventArgs e)
        {
           RI.SaveDocumentStatus(Int64.Parse(txtDocId.Text), 3,0);
        }
    }
}